include/rrc/io/IOService.h

```diff
@@ -0,0 +1,26 @@
+/**
+ *  @autor nikitas
+ *  @date 10/7/16
+ */
+#pragma once
+
+#include <queue>
+#include <functional>
+
+#include "Definitions.h"
+
+namespace rrc
+{
+    typedef std::function<void()> AsyncCommand;
+
+    class IOService
+    {
+    public:
+        void run();
```

Comment:
This class should have method bool step(). run() method should call step and have an opportunity to stop execution.